docs/PORTABLE_SERVICES.md

```diff
@@ -243,6 +250,34 @@ image. To facility 3 and 4 you also need to include a boot loader in the
 image. As mentioned `mkosi -b` takes care of all of that for you, but any other
 image generator should work too.
 
+## Layered Images
+
+Portable services can be delivered as multiple images that get layered and
+combined with OverlayFS at runtime, when they are attached. This enables a
+workflow that splits the base 'runtime' from the daemon, so that multiple
+portable services can share the same 'runtime' image (libraries, tools)
+without having to include everything each time. The `--extension` parameter
+of `portablectl` can be used to specify as many upper layers as desired.
+On top of the requirements listed in the previous section, the following
+must be also be observed.
+
+1. The base/OS layer image must contain an os-release file, either in `/etc/os-release` or
+   `/usr/lib/os-release`. The file should follow the standard format.
+
+2. The upper layer(s) image(s) must contain an extension-release file in
+   `/usr/lib/extension-release.d/`, with an `ID=` and `SYSEXT_LEVEL=`/`VERSION_ID=`
+   matching the base layer.
+
+3. The base/OS layer image does not need to have any unit files.
```

Comment:
hmm, referencing unit files is a bit problematic, as right now systemd-sysext runs after PID 1 already loaded its units, hence they cannot be used to make unit files available to the OS. we should do something about that though. but maybe we shouldn't mention this here for now.

Reply:
But this document is about portable services, not systemd-sysext - the clarification is needed because in the standard/monolithic mode in the section above (`Requirements on Images`) we say explicitly that the image must have a unit file. Hence here I am clarifying that in the layered mode, the base layer doesn't have to contain any. Given this is a specification I think it's worth being precise. And there shouldn't be any confusion with systemd-sysext, although the implementation uses the same code, the feature is completely separate and independent, I don't think there can be any confusion?

Reply:
ok